apps/laboratory/src/utils/BitcoinUtil.ts

```diff
@@ -0,0 +1,141 @@
+import type { BitcoinConnector } from '@reown/appkit-adapter-bitcoin'
+import type { CaipNetwork, CaipNetworkId } from '@reown/appkit'
+import * as networks from '@reown/appkit/networks'
+import * as bitcoin from 'bitcoinjs-lib'
+
+export const BitcoinUtil = {
+  createSignPSBTParams(params: BitcoinUtil.CreateSignPSBTParams): BitcoinConnector.SignPSBTParams {
+    const network = this.getBitcoinNetwork(params.network.caipNetworkId)
+
+    const psbt = new bitcoin.Psbt({ network })
+    const payment = bitcoin.payments.p2wpkh({ address: params.senderAddress, network })
+
+    if (!payment.output) {
+      throw new Error('Invalid payment output')
+    }
+
+    const change = this.calculateChange(params.utxos, params.amount, params.feeRate)
+
+    if (change < 0) {
+      throw new Error('Insufficient funds')
+    } else if (change > 0) {
+      psbt.addOutput({
+        address: params.senderAddress,
+        value: BigInt(change)
+      })
+    }
+
+    for (const utxo of params.utxos) {
+      psbt.addInput({
+        hash: utxo.txid,
+        index: utxo.vout,
+        witnessUtxo: {
+          script: payment.output,
+          value: BigInt(utxo.value)
+        }
+      })
+    }
+
+    psbt.addOutput({
+      address: params.recipientAddress,
+      value: BigInt(params.amount)
+    })

```

Comment:
we'll probably need this internally on the wallet standard side of things right?

Reply:
Yeah you are correct, we will probably need that in WalletStandard to create the transfer functionality. I think we will need to get it differently as we are relying into an API to get the UTXO and fees, so we need to check which would be the best approach to have this internally.